tests/test_kernel.py

```diff
@@ -105,8 +105,8 @@ def child_target():
             sleep({interval})
 
     def parent_target():
-        sleep({interval})
         Thread(target=child_target).start()
+        sleep({interval * iterations})
```

Comment:
Does it also work if in addition to this sleep you keep the `sleep({interval})` above? I wonder if that was here for a reason. In other words, I think that adding the sleep at the end is fine, I am less confident about the function of the sleep that was removed (as this can change some edge case related to execution order).

Further, should we save a reference to the thread in a local variable before calling `sleep()`?

Reply:
In https://github.com/ipython/ipykernel/pull/1312/commits/41b31c5ca070bae713d80b15739011143ac58483 I added back the first `sleep`, and replaced the other `sleep` with a `thread.join()`, since it's essentially the same.